network.go

```diff
@@ -693,6 +693,11 @@ func setupDNS(dns []string) (err error) {
 		return nil
 	}
 
+	if _, err := os.Stat(guestDNSFile); err != nil && os.IsNotExist(err) {
```

Comment:
Guest DNS is setup only if the grpc request contains dns bits asking for guest dns to be setup. I feel in this case. since an explicit request has been made to setup dns, we should be creating the /etc/resolve.conf file so that `kataGuestSandboxDNSFile` can be bind-mounted to it. While `/etc/resolve.conf` file  exists by default on most distros, this file is absent on Clear Linux distribution as it is stateless. I have a feeling the original change to set up dns was not tested a Clear Linux based root file system.

cc @bergwolf @lifupan

Reply:
IMHO, most of users wouldn't care about the guest DNS setting, and we wouldn't want this setting error disturbing those users who didn't care about it.  For those users who did need the guest DNS setting, they should make sure the "/etc/resolve.conf" file exists in their custom guest image.

Reply:
Fair enough. This is something we could handle while generating the clearlinux rootfs.